src/agent/onefuzz-supervisor/src/work.rs

```diff
@@ -113,45 +115,42 @@ impl IWorkQueue for WorkQueue {
         self.poll().await
     }
 
-    async fn claim(&mut self, receipt: Receipt) -> Result<()> {
-        self.claim(receipt).await
+    async fn claim(&mut self, message: Message) -> Result<WorkSet> {
+        self.claim(message).await
     }
 }
 
 impl_downcast!(IWorkQueue);
 
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Debug)]
 pub struct Message {
-    pub receipt: Receipt,
+    pub queue_message: Option<QueueMessage>,
     pub work_set: WorkSet,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct Receipt(pub storage_queue::Receipt);
-
 pub struct WorkQueue {
     queue: QueueClient,
-    registration: Registration,
+    _registration: Registration,
```

Comment:
why is this _registration?

Reply:
That was the result of some experimentation, will rename